### DISCUSSION TITLE
Invite (Password provider)

### DISCUSSION THREAD

**Discussion by SebastianPfliegel** [nuxt-community/supabase-module#439] [Q&A]
In my application I only want invited users to be able to login. Users should not be able to sign up, because every user is an admin/editor. The base (`/`) route should be available publicly. `/admin` should only be visible for authenticated users. `/admin/login` should be the login form. After the user clicked the invitation link I want the user set an initial password (maybe on the route `/admin/register`). @nuxtjs/supabase expects some redirection settings. I've set them up like this:

```js
supabase: {
    redirectOptions: {
      login: '/admin/login',
      callback: '/admin/confirm',
      include: ['/admin(*)?']
    },
  },
```

I don´t know what callback is used for in my case.

As far as I know the user should be automatically logged in after clicking the invitation link, which is unfortunately not the case. Any ideas why clicking the invitation link does not log me in?

**Comment by yakubka** [comment]
For an invite-only flow where users cannot self-register you need to combine three things: disable signups in Supabase Auth settings, send invites from the server, and handle the invite callback.

Step 1. Disable public signups. In your Supabase dashboard go to Authentication > Settings and turn off "Enable email signup". This blocks the public signUp() call.

Step 2. Send invites from a server route using the service role client:

// server/api/admin/invite.post.ts
import { serverSupabaseServiceRole } from "#supabase/server"

export default defineEventHandler(async (event) => {
  const client = serverSupabaseServiceRole(event)
  const { email } = await readBody(event)
  const { error } = await client.auth.admin.inviteUserByEmail(email, {
    redirectTo: "https://yoursite.com/admin/confirm",
  })
  if (error) throw createError({ statusCode: 400, message: error.message })
  return { ok: true }
})

Step 3. Handle the invite confirmation on /admin/confirm. The user lands there with a token_hash and type=invite in the query string. The modules built-in callback route (configured via redirectOptions.callback) already exchanges the code, so you only need to redirect the user to your password-set page after the session is established. Set redirectOptions.callback to /admin/confirm and redirectOptions.login to /admin/login in your nuxt.config.ts.
